### PR TITLE
snap, client, daemon, store: use and expose "media" more

### PR DESCRIPTION
**Pull request by chipaca**
Upcoming GUI work needs access to the "media" property from the
store. I'd hoped to be able to delay this until we'd dropped v1 (and
thus screenshots), but they need it RSN, and dropping v1 is looking
like a six-month thing. This exposes "media" straight to the client,
and populates screenshots and icon url from it instead of requesting
them individually.

### CODE REVIEW

File: daemon/snap.go [hunk 1]
```diff
@@ -415,7 +406,8 @@ func mapRemote(remoteSnap *snap.Info) *client.Snap {
 		Contact:      remoteSnap.Contact,
 		Title:        remoteSnap.Title(),
 		License:      remoteSnap.License,
-		Screenshots:  screenshots,
+		Screenshots:  remoteSnap.Media.Screenshots(),
```
Comment (zyga) [inline daemon/snap.go]: Oooh, nice :-)

File: snap/info.go [hunk 2]
```diff
@@ -743,9 +744,46 @@ type AppInfo struct {
...
+}
+
+type MediaInfo struct {
+	Type   string `json:"type"`
+	URL    string `json:"url"`
+	Width  int64  `json:"width,omitempty"`
+	Height int64  `json:"height,omitempty"`
+}
+
+func (mi MediaInfo) Screenshot() ScreenshotInfo {
```
Comment (pedronis) [inline snap/info.go]: this, as is, is strange given that it doesn't check type

File: snap/info.go [hunk 3]
```diff
@@ -244,6 +244,7 @@ type Info struct {
 	Publisher StoreAccount
 
 	Screenshots []ScreenshotInfo
```
Comment (pedronis) [inline snap/info.go]: why not remove this as well now?

File: snap/info.go (hunk 2 above)
Comment (chipaca) [reply snap/info.go]: maybe i should make it private (it's just a helper for `Screenshots()`)

**Pull request merged**